src/libcpu/src/jit/jit_pairedsingle.cpp

```diff
@@ -23,23 +49,30 @@ mergeGeneric(PPCEmuAssembler& a, Instruction instr)
       return jit_fallback(a, instr);
    }
 
-   // We need to use a temporary here since frA/frB and frD may overlap
-   auto srcA = a.loadRegisterRead(a.fprps[instr.frA]);
-   auto tmpSrcB = a.allocXmmTmp(a.loadRegisterRead(a.fprps[instr.frB]));
-
-   uint8_t shflFlags = 0;
-
-   if (flags & MergeValue0) {
-      shflFlags |= (1 << 0);
-   }
-
-   if (flags & MergeValue1) {
-      shflFlags |= (1 << 1);
+   auto tmpSrcA = a.allocXmmTmp();
+   auto tmpSrcB = a.allocXmmTmp();
+   {
+      auto srcA = a.loadRegisterRead(a.fprps[instr.frA]);
+      auto srcB = a.loadRegisterRead(a.fprps[instr.frB]);
+      if (flags & MergeValue0) {
+         a.movapd(tmpSrcA, srcA);
+         a.shufpd(tmpSrcA, tmpSrcA, 1);
+         roundToSingleSd(a, tmpSrcA, tmpSrcA);
```

Comment:
I do not believe that the interpreter truncates or rounds the values, why are we doing it here.